File: PS_java/Bj7570.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		
		int[] nums = new int[N];
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i=0; i<N; i++) {
			nums[Integer.parseInt(st.nextToken())-1] = i;
			
		}
		int temp = 1;
		int ans = 1;
		for (int i=1; i<N; i++) {
			if (nums[i] > nums[i-1]) temp++;
			else temp = 1;
			ans = Math.max(ans, temp);
		}
		
		System.out.println(N-ans);
		
	}
}

File: Baekjoon/Bj7570.py
# 7570 줄세우기


def solution():
    N = int(input())
    nums = [0] * N
    for i, n in enumerate(map(int, input().split())):
        nums[n - 1] = i
    temp = 1
    ans = 1
    for i in range(1, N):
        if nums[i] > nums[i - 1]:
            temp += 1
        else:
            temp = 1
        ans = max(ans, temp)
    return N - ans


if __name__ == "__main__":
    print(solution())
